src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,242 @@
+    using MevMomentum = units::MevMomentum;
+    using Mass = units::MevMass;
+    using UniformRealDist = UniformRealDistribution<real_type>;
+    //!@}
+
+    // Construct with shared and state data
+    inline CELER_FUNCTION
+    MuDecayInteractor(MuDecayData const& shared,
+                      ParticleTrackView const& particle,
+                      Real3 const& inc_direction,
+                      StackAllocator<Secondary>& allocate);
+
+    // Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Constant data
+    MuDecayData const& shared_;
+    // Incident muon energy
+    Energy const inc_energy_;
+    // Incident muon direction
+    Real3 const& inc_direction_;
+    // Allocate space for secondary particles (electron only)
+    StackAllocator<Secondary>& allocate_;
+    // Define decay channel based on muon or anti-muon primary
+    bool is_muon_;
+    // Incident muon four vector
+    FourVector inc_fourvec_;
+    // Max sampled fractional energy for the electron neutrino (not unity)
+    UniformRealDist sample_electron_nu_;
+    // Maximum electron energy [MeV]
+    real_type max_energy_;
+
+    //// HELPER FUNCTIONS ////
+
+    // Boost four vector from the rest frame to the lab frame
+    inline CELER_FUNCTION FourVector to_lab_frame(Real3 const& dir,
+                                                  MevMomentum momentum,
+                                                  Mass mass) const;
+
+    // Calculate particle momentum (or kinetic energy) in the center of mass
+    inline CELER_FUNCTION MevMomentum calc_momentum(real_type energy_frac,
+                                                    Mass mass) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with shared and state data.
+ *
+ * \note Geant4 physics manual defines E_{max} = m_\mu / 2, while the source
+ * code (since v10.2.0 at least) defines E_{max} = m_\mu / 2 - m_e . The source
+ * code implementation leads to a total CM energy of ~104.6 MeV instead of the
+ * expected 105.7 MeV (muon mass), which is achieved by using the physics
+ * manual definition.
+ */
+CELER_FUNCTION
+MuDecayInteractor::MuDecayInteractor(MuDecayData const& shared,
+                                     ParticleTrackView const& particle,
+                                     Real3 const& inc_direction,
+                                     StackAllocator<Secondary>& allocate)
+    : shared_(shared)
+    , inc_energy_(particle.energy())
+    , inc_direction_(inc_direction)
+    , allocate_(allocate)
+    , is_muon_(particle.particle_id() == shared_.ids.mu_minus)
+    , inc_fourvec_({inc_direction_ * particle.momentum().value(),
+                    particle.total_energy().value()})
```

Comment:
should be
```suggestion
    , inc_fourvec_{inc_direction_ * value_as<MevMomentum>(particle.momentum()),
                    value_as<MevEnergy>(particle.total_energy())}
```
to avoid a copy and to guarantee the consistency between the units